src/microk8s.py

```diff
@@ -0,0 +1,110 @@
+#
+# Copyright 2023 Canonical, Ltd.
+#
+import json
+import logging
+import os
+import subprocess
+from pathlib import Path
+from typing import Optional
+
+from ops.model import ActiveStatus, MaintenanceStatus, WaitingStatus
+
+import util
+
+LOG = logging.getLogger(__name__)
```

Comment:
You could consider a class here.

Reply:
I started with putting this on a class, but it was way harder to setup unit test mocks for the charms afterwards (due to the way that the harness initializes the charm itself).